### PR TITLE
Update lint documentation to use markdown headlines

### PR DESCRIPTION
**Pull request by xFrednet**
This PR updates all lint documentation to use markdown headlines. It additionally removed the *Known problems* section for lints without any problems. I've double-checked all automatic replacements, but a second pair of eyes is definitely appreciated!

I wasn't sure when you wanted to switch to the new metadata collection tomorrow, I therefore prepared this PR today. And that's it this is a standalone PR to keep the other related PRs reviewable. 

changelog:  none

r? @flip1995 

cc: #7172 

Note: This should be merged with the other metadata collection related PRs.

### CODE REVIEW

**Comment by flip1995** [comment]
> seems like `r?` isn't working currently thinking

Yeah, it seems GitHub recently has problems with webhooks, so the bots are a bit flaky. 


---

As you may have noticed: I didn't get to the move this weekend and try to fit it in during this week.

**Comment by xFrednet** [comment]
I'm actually surprised it took this long xD. I'll update the PR in the next couple of days (hopefully tomorrow)

**Comment by camsteffen** [comment]
self_named_constructor lint was added

**Comment by xFrednet** [comment]
Thank you for the ping! I'll update this PR tomorrow :upside_down_face:

File: doc/adding_lints.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ because that's clearly a non-descriptive name.
   - [Setup](#setup)
   - [Getting Started](#getting-started)
   - [Testing](#testing)
+    - [Cargo lints](#cargo-lints)
```
Comment (flip1995) [inline doc/adding_lints.md:14]: Ninja fix :smile:

**Review by flip1995 — APPROVED** [review]
LGTM overall. This messes up the formatting of the in-code documentation a bit (line length of the first doc line) but we weren't consistent in that regard anyway. Also it won't have any influence on the website.

I'll merge this after the move.

File: doc/adding_lints.md (hunk 1 above)
Comment (xFrednet) [reply doc/adding_lints.md:14]: xD even for me. A IDE plugin did that and I only noticed it when I reviewed the changes on GitHub xD

I would also suggest merging this last, another idea I had was to crate a rollup PR that combines all metadata related PRs :thinking: You'll find a good solution, I have faith in you :upside_down_face:

**Pull request merged**